pkgs/development/python-modules/mesa/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, isPy27
+, cookiecutter, networkx , pandas, tornado, tqdm
+, pytest, pytestCheckHook }:
+
+buildPythonPackage rec {
+  pname = "mesa";
+  version = "0.8.6";
+
+  # According to their docs, this library is for Python 3+.
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    repo = "mesa";
+    owner = "projectmesa";
+    rev = "v${version}";
+    sha256 = "0d8c636zhswxd91ldlmdxxlyym2fj3bk1iqmpc1jp3hg7vvc7w03";
+  };
+
+  # Disable tests of the examples because of import errors (due to directory
+  # layout?).
+  preCheck = ''
+    cat > conftest.py <<EOF
+    import sys
+    collect_ignore_glob = ["examples/*"]
+    EOF
+  '';
+
+  checkInputs = [ pytest pytestCheckHook ];
```

Comment:
whn adding the hook we also need to pass `dontUseSetuptoolsCheck = true;` as it will otherwise try to run two phases.